### ISSUE TITLE
source_catalog ValueError comparing micro_Jy

### ISSUE THREAD

**Issue by jdavies-st** [STScI-JWST/jwst#976]
@larrybradley 

Seeing a problem on [line 155](https://github.com/STScI-JWST/jwst/blob/master/jwst/source_catalog/source_catalog.py#L155) in `make_source_catalog`.  In the previous block, if the `else` clause gets executed, then the comparison on line 155 becomes a comparison between an array and 0.0.  The traceback:
```
Traceback (most recent call last):
  File "/Users/jdavies/dev/jwst/scripts/strun", line 14, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 142, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/cmdline.py", line 320, in step_from_cmdline
    step.run(*positional)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 356, in run
    result = self.process(*args)
  File "/Users/jdavies/dev/jwst/jwst/pipeline/calwebb_image3.py", line 97, in process
    out_catalog = self.source_catalog(output)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 356, in run
    result = self.process(*args)
  File "/Users/jdavies/dev/jwst/jwst/source_catalog/source_catalog_step.py", line 42, in process
    npixels, deblend=deblend)
  File "/Users/jdavies/dev/jwst/jwst/source_catalog/source_catalog.py", line 155, in make_source_catalog
    if micro_Jy > 0.0:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

The MIRI imaging test data that is throwing this error has the following `meta.photometry` block:
```
{'conversion_megajanskys': 2.449465751647949,
 'conversion_microjanskys': 57.57329629700661,
 'pixelarea_arcsecsq': 0.0121,
 'pixelarea_steradians': 2.84403609523084e-13}
```
Not sure how this block gets populated.  Other imaging test data I have from NIRCam doesn't have it and doesn't throw this error.

**Comment by hbushouse** [comment]
The conversion_nnn and pixelarea_nnn meta entries get populated by the PHOTOM step at the end calimage2/calspec2 processing. They correspond to keywords PHOTMJSR (conversion factor to put data into units of megaJy/steradian) and PHOTUJA2 (conversion factor to put data into units of microJy/arcsec^2). I'm surprised you aren't seeing them populated in the simulated NIRCam images. They get loaded from the photom reference file based on the detector, filter, and pupil in use. Perhaps the filter+pupil settings in the NIRCam sim data are off-nominal, such that there aren't any entries in the photom ref file for them.

**Comment by jdavies-st** [comment]
Thanks Howard.  I will have a look at the sim data and compare to the logic in the photom step.  I bet that's the issue.

**Comment by jdavies-st** [comment]
So it appears that the [a block](https://github.com/STScI-JWST/jwst/blob/master/jwst/photom/photom.py#L666-L683) on the `save_area_info` method checks to see if the area reference file and the photometry reference file have the same values of the following keywords:
```
meta.photometry.pixelarea_arcsecsq
meta.photometry.pixelarea_steradians
```
If they do not (within a tolerance), it doesn't write them to the output `_cal` file.

It turns out the PHOTOM reffile for NIRCam has these keywords set to zero, whereas the AREA reffile has what seem like real values.  Since they are different, they don't get written to the output `_cal` file and thus don't cause problems for `source_catalog`.

**Comment by hbushouse** [comment]
Another victim of instrument teams delivering ad hoc, poorly populated reference files early in the development phase just to get something in there. I always thought that having the pixel area stored in two different reference files was a recipe for trouble. I know the pixel area figures into the calibration data stored in the photom ref files, but it's most naturally associated with the pixel area reference file and hence should only be stored there. I guess we need to contact our friend Bryan and ask if the NIRCam team can deliver updated photom ref files that have the pixel area keywords populated with non-zero values. Just copy the values from the pixel area ref files.

**Comment by larrybradley** [comment]
I completely agree that this value should be stored in only one place.  You're asked for trouble otherwise.

**Comment by jdavies-st** [comment]
If we agree that pixel area should be only in the AREA reffile, should we updated the `photom` code to only look for it there? And also change the schema for the PHOTOM reffiles to not include pixel area?

We of course can update the code before new reffiles are released.  I believe the schema for the reffiles can even be changed and the old reffiles will still be read in OK - the pixel area entries will just be shuffled off to `_extra_fits`.

**Comment by hbushouse** [comment]
We can certainly make both those updates (the photom step code and the ref file data model). But before we do, I should verify with the DMS WG that this is an OK plan of action. In the meantime I suppose we could at least just comment out that block of the photom step code that does the comparison, so that it always writes the values to the output, just to get things to work in the meantime.

**Comment by hbushouse** [comment]
Original problem fixed in #983. Still need to update photom step docs to reflect the new behavior.
